src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -1010,8 +1010,22 @@ private List<AssociativeNode> GetModifiedNodes(Subtree subtree, bool redefinitio
                     // It can then be handled normally regardless of its ForceExecution state
                     subtree.ForceExecution = false;
 
-                    //Check if the subtree (ie a node in graph) is an input and has primitive Right hand Node type
-                    if (redefinitionAllowed && st.IsInput && node is BinaryExpressionNode bne  && CoreUtils.IsPrimitiveASTNode(bne.RightNode))
+                    BinaryExpressionNode prevBNE = null;
+                    var bne = node as BinaryExpressionNode;
+                    if (bne != null){
+                        foreach (var astNode in st.AstNodes)
+                        {
+                            if(astNode is BinaryExpressionNode curBNE && curBNE.LeftNode is IdentifierNode id && id.Name == bne.LeftNode.Name)
+                            {
+                                prevBNE = curBNE;
+                            }
+                        }
+                    }
+
+                    //Check if the subtree (ie a node in graph) is an input and has primitive Right hand Node type, also
+                    //ensure that the previous RHS of this assignment was a primitive value, or we'll have generated incorrect instructions.
+                    if (redefinitionAllowed && st.IsInput && bne != null && CoreUtils.IsPrimitiveASTNode(bne.RightNode)
+                        && (CoreUtils.IsPrimitiveASTNode(prevBNE.RightNode) || prevBNE == null))
```

Comment:
I thought the condition `prevBNE == null` needed to precede the second condition invoking the property on it, but I guess the compiler is smart enough to decide the order of evaluating the conditions :)

Reply:
well, just in case, I'll fix that.